elixir/apps/domain/lib/domain/resources.ex

```diff
@@ -46,6 +47,35 @@ defmodule Domain.Resources do
     end
   end
 
+  def count_resources_for_gateway(%Gateways.Gateway{} = gateway, %Auth.Subject{} = subject) do
+    required_permissions =
+      {:one_of,
+       [
+         Authorizer.manage_resources_permission(),
+         Authorizer.view_available_resources_permission()
+       ]}
+
+    with :ok <- Auth.ensure_has_permissions(subject, required_permissions) do
+      count =
+        Resource.Query.all()
+        |> Authorizer.for_subject(subject)
+        |> Resource.Query.by_gateway_group_id(gateway.group_id)
+        |> Repo.aggregate(:count)
+
+      {:ok, count}
+    end
+  end
+
+  def list_resources_for_gateway(%Gateways.Gateway{} = gateway, %Auth.Subject{} = subject) do
+    resources =
```

Comment:
Sorry for not leaving a comment here, it also needs all fixes you applied to the function above. Just as a rule of thumb:
1. If the function is called from a front-end or API (not by internal logic) it must have a subject;
2. If it has a subject - always check it's permissions;
3. If it does a query - always filter by `for_subject`.

I think we should write it somewhere down but not sure what the place would be for such conventions.